x-pack/plugins/transform/server/lib/alerting/transform_health_rule_type/register_transform_health_rule_type.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import type {
+  ActionGroup,
+  AlertInstanceContext,
+  AlertInstanceState,
+  AlertTypeState,
+} from '../../../../../alerting/common';
+import { PLUGIN, TRANSFORM_RULE_TYPE } from '../../../../common/constants';
+import { transformHealthRuleParams, TransformHealthRuleParams } from './schema';
+import { AlertType } from '../../../../../alerting/server';
+import { transformHealthServiceProvider } from './transform_health_service';
+
+export interface BaseResponse {
+  transform_id: string;
+  description?: string;
+}
+
+export interface NotStartedTransformResponse extends BaseResponse {
+  transform_state: string;
+  node_name?: string;
+}
+
+export type TransformHealthResult = NotStartedTransformResponse;
+
+export type TransformHealthAlertContext = {
+  results: TransformHealthResult[];
+  message: string;
+} & AlertInstanceContext;
+
+export const TRANSFORM_ISSUE = 'transform_issue';
+
+export type TransformIssue = typeof TRANSFORM_ISSUE;
+
+export const TRANSFORM_ISSUE_DETECTED: ActionGroup<TransformIssue> = {
+  id: TRANSFORM_ISSUE,
+  name: i18n.translate('xpack.transform.alertingRuleTypes.transformHealth.actionGroupName', {
+    defaultMessage: 'Issue detected',
+  }),
+};
+
+export function getTransformHealthRuleType(): AlertType<
+  TransformHealthRuleParams,
+  never,
+  AlertTypeState,
+  AlertInstanceState,
+  TransformHealthAlertContext,
+  TransformIssue
+> {
+  return {
+    id: TRANSFORM_RULE_TYPE.TRANSFORM_HEALTH,
+    name: i18n.translate('xpack.transform.alertingRuleTypes.transformHealth.name', {
+      defaultMessage: 'Transform health',
+    }),
+    actionGroups: [TRANSFORM_ISSUE_DETECTED],
+    defaultActionGroupId: TRANSFORM_ISSUE,
+    validate: { params: transformHealthRuleParams },
+    actionVariables: {
+      context: [
+        {
+          name: 'results',
+          description: i18n.translate(
+            'xpack.transform.alertTypes.transformHealth.alertContext.resultsDescription',
+            {
+              defaultMessage: 'Results of the rule execution',
```

Comment:
It's a bit shorter this way.
```suggestion
              defaultMessage: 'Rule execution results',
```

Reply:
Updated in https://github.com/elastic/kibana/pull/112277/commits/2d3db4d82c480a2fafe7c08238d10af3b996fe85